oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthCodeHandler.java

```diff
@@ -190,16 +187,78 @@ public APIGatewayProxyResponseEvent handleRequest(
                                     authorisationCodeService.generateAuthorisationCode(
                                             sessionCookieIds.getClientSessionId(),
                                             session.getEmailAddress());
-                            AuthenticationSuccessResponse authenticationResponse =
-                                    authorizationService.generateSuccessfulAuthResponse(
-                                            authenticationRequest, authCode);
-                            sessionService.save(session.setState(nextState));
-                            return new APIGatewayProxyResponseEvent()
-                                    .withStatusCode(302)
-                                    .withHeaders(
-                                            Map.of(
-                                                    ResponseHeaders.LOCATION,
-                                                    authenticationResponse.toURI().toString()));
+
+                            try {
+                                String clientRegistryCookieConsentValue =
+                                        authorizationService.getClientCookieConsentValue(
+                                                authenticationRequest.getClientID());
+                                String cookieConsentParamValue =
+                                        getCookieConsentParamValue(
```

Comment:
Should we only ever do this check and pass the COOKIE_CONSENT_PARAM_NAME  in the auth response if the CookieConsent is true? and if it is false we can just ignore and continue?

Reply:
updated to only add the params when CookieConsent is true